services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/APISurfaceAreaReview.java

```diff
@@ -0,0 +1,401 @@
+        CustomClassForDocumentAPI customAttr = customDoc.get("custom_attr", EnhancedType.of(CustomClassForDocumentAPI.class));
+        System.out.println("customAttr " +customAttr);
+        assertThat(customAttr).isEqualTo(customObject);
+
+
+        // Extracting custom List
+        List<CustomClassForDocumentAPI> extractedList = customDoc.getList("customO_list",
+                                                                        EnhancedType.of(CustomClassForDocumentAPI.class));
+
+        assertThat(extractedList).isEqualTo(Arrays.asList(customObjectOne,customObjectTwo));
+
+    }
+
+
+    /**
+     * ERROR Case when Attribute Converters are not supplied and a custom Object is added
+     */
+    @Test
+    void attributeConverter_Not_Added(){
+
+        CustomClassForDocumentAPI customObject = getCustomObject("str", 25L, false);
+
+        assertThatExceptionOfType(IllegalStateException.class)
+            .isThrownBy(() ->
+                EnhancedDocument.builder()
+                                .addString("direct_attr", "sample_value")
+                                .add("custom_attr", customObject)
+                                .build()
+
+            ).withMessage("Converter not found for EnhancedType(software.amazon.awssdk.enhanced.dynamodb.converters"
+                                      + ".document.CustomClassForDocumentAPI)");
+
+    }
+
+
+    @Test
+    void putSimpleDocumentWithSimpleValues() {
+        table.putItem(EnhancedDocument
+                          .builder()
+                          .addString(HASH_KEY, "first_hash")
+                          .addString(SORT_KEY, "1_sort")
+                          .addNull("null_Key")
+                          .addBoolean("boolean_key", true)
+                          .addNumber("number_int", 2)
+                          .addNumber("number_SdkNumber", SdkNumber.fromString("5"))
+                          .addSdkBytes("SdkBytes", SdkBytes.fromUtf8String("sample_binary"))
+                          .build());
+
+        assertThat(
+            table.getItem(EnhancedDocument.builder()
+                                          .add(SORT_KEY, "1_sort")
+                                          .addString(HASH_KEY, "first_hash").build()).get(SORT_KEY)).isEqualTo("1_sort");
+
+
+    }
+
+    /**
+     *- Does EnhancedDocument.Builder needs add(key, Object, attributeConverterProvider)
+     *
+     * - Should we refer bytes as binary or byte in apis example addBinary(byte[] byte) or addByte(byte[] byte)
+     *
+     * - Map<String, Object> getRawMap(String attributeName); ==> Can we remove the use of Object ? <Zoe>
+     *
+     * - getJsonPretty. Do we actually need this?
+     *
+     * - Object get(String attributeName); ==> Do we need this
+     * -  We don't have jsonPretty utility elsewhere. Do we need it?
+     *
+     * - Builder add(String attributeName, Object value); ==> Using raw type is a bit code smell and error prone to me.Should  it
```

Comment:
+1, `add(String attributeName, Object value)` is not type safe and would defeat the purpose of all other methods IMO

Reply:
we might need it for adding beans, though, where we don't know the type but we have a configured attribute converter.

Reply:
Yeah, just realized it now 😅 . I wonder how we can prevent people from using `add(String attributeName, Object value)` for everything. Maybe we can rename it to something like `addCustomType`?

Reply:
putGeneric() ? putCustom() ? poll
putObject